backend/remote/remote.go

```diff
@@ -280,6 +281,7 @@ func (rf *Factory) SignalToAdd(address string, action string) error {
 			Timeout: timeout,
 		},
 	}
+	time.Sleep(inject.SignalToAddTimeout)
```

Comment:
This is ok. However, do think about other ways to inject.

How about creating a new Factory and making use of this new Factory. The original Factory remains unchanged. The enhancements & or fixes go into original Factory only. Injection related capabilities are isolated in this new Factory called `injectibleFactory`.

```go
func InjectibleFactory() types.BackendFactory {
  return &InjectingFactory{}
}

type InjectingFactory struct {
  Factory
}

// verify if InjectingFactory implements all the methods exposed by 
// types.BackendFactory
var _ types.BackendFactory = &InjectingFactory{}

func (if *InjectingFactory) SignalToAdd(address string, action string) error {
  time.Sleep(inject.SignalToAddTimeout)
  if.Factory.SignalToAdd(address, action)
}
```

Reply:
In addition, the image tag can be `0.6.0-debug` & so on.

Reply:
Thanks @AmitKumarDas  for the review, will change the image tag.